### ISSUE TITLE
Bug or feature? Does LTE eNodeB fallback exist?

### ISSUE THREAD

**Issue by Vitaly-13** [mozilla/ichnaea#507]
I have noted that service returns data for non existing cells for some LTE eNodeBs.
Examle for 10 cells for eNodeB with mcc=250,mnc=99,tac=22977, eNodeBId=772724:
CID=0 (eci=197817344): loc [55.9411657 , 37.493728] accur 1000.0
CID=1 (eci=197817345): loc [55.9411657 , 37.493728] accur 1000.0
CID=2 (eci=197817346): loc [55.9411657 , 37.493728] accur 1000.0
CID=3 (eci=197817347): loc [55.9411657 , 37.493728] accur 1000.0
CID=4 (eci=197817348): loc [55.9419916 , 37.4937153] accur 1000.0
CID=5 (eci=197817349): loc [55.9399009 , 37.498864] accur 1000.0
CID=6 (eci=197817350): loc [55.941039 , 37.4914616] accur 1000.0
CID=7 (eci=197817351): loc [55.9411657 , 37.493728] accur 1000.0
CID=8 (eci=197817352): loc [55.9411657 , 37.493728] accur 1000.0
CID=9 (eci=197817353): loc [55.9411657 , 37.493728] accur 1000.0

Real cells have CIDs: 4,5,6. Other cells, which do not exist, have same location data.
Is it some kind of fallback or bug?
If this is fallback, is it possible to switch off this fallback?

**Comment by hannosch** [comment]
How do you query the API? Do you provide the eci as the cellId field in the request? The cell id field is supposed to be the full 28 bit cell identity for LTE networks (https://developer.android.com/reference/android/telephony/CellIdentityLte.html#getCi()).

We don't actually look at, compute or take in the eNodeBId in the API or backend code as a distinct value at all. So if we have those return values and you already switched off ipf and lacf fallbacks, it means client devices out there have reported those cell identity values to us. We don't really have any way to trust some of these over others.

**Comment by Vitaly-13** [comment]
Yes, I provide ECI as cell ID.
I have found in export file that cells. You may see below.
It seems that I know how it is possible.
What will happen if in one location request there will be several cells including known and fake cells? Will service add that unknown fake cells to database with some average coordinates?
 
LTE,250,99,22977,197817344,,37.4937280,55.9411657,56,32,1,1494006011,1494954144,
LTE,250,99,22977,197817345,,37.4937280,55.9411657,56,31,1,1494006011,1494954144,
LTE,250,99,22977,197817346,,37.4937280,55.9411657,56,31,1,1494006011,1494954144,
LTE,250,99,22977,197817347,,37.4937280,55.9411657,56,30,1,1494006011,1494954144,
LTE,250,99,22977,197817348,480,37.4937153,55.9419916,68,6,1,1489590773,1489796537,
LTE,250,99,22977,197817349,481,37.4988640,55.9399009,631,7,1,1422719375,1484326953,
LTE,250,99,22977,197817350,44,37.4914616,55.9410390,345,21,1,1482444739,1484326953,
LTE,250,99,22977,197817351,,37.4937280,55.9411657,56,3,1,1494006011,1494006093,
LTE,250,99,22977,197817352,,37.4937280,55.9411657,56,3,1,1494006011,1494006093,
LTE,250,99,22977,197817353,,37.4937280,55.9411657,56,3,1,1494006011,1494006093,

**Comment by hannosch** [comment]
Yes, you are correct. We are using the data from queries to extend the dataset. So if we know the position of cell A and we see a query containing cell A and a previously unknown cell B, we assume cell B is initially at the same position as cell A, with a large accuracy radius around it.

But this isn't really different from someone providing us with a GPS based position and claiming cell A and B are both at the same place and one or both of them having incomplete cell identities.

If some cell modems provide us with incomplete cell identifiers, they'll do that both as part of queries and as part of GPS based "ground truth" stumbling data, as both are likely using the same "get neighboring cells" or "get all cells" APIs from the operating system.
